[CI] PreallocateTests testPreallocateNonExistingDirectory failing

**Build Scans:**
- [elasticsearch-periodic #3703 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/ax7uk3gcofcyu)
- [elasticsearch-periodic #3699 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/uvhpitu2xcktw)
- [elasticsearch-periodic #3695 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/3nmwqdv6yw6vq)
- [elasticsearch-periodic #3691 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/qgv5hjfjliqpc)
- [elasticsearch-periodic #3686 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/kqrdkr34tqu54)
- [elasticsearch-periodic #3682 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/altttp3bk2rxy)

**Reproduction Line:**
```
./gradlew ":libs:elasticsearch-native:test" --tests "org.elasticsearch.nativeaccess.PreallocateTests.testPreallocateNonExistingFile" -Dtests.seed=FB3FF84515445BCD -Dtests.locale=sr-Latn-RS -Dtests.timezone=Asia/Katmandu -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testPreallocateNonExistingDirectory'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.nativeaccess.PreallocateTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <1048576L>
     but: was <0L>
```

**Issue Reasons:**
- [main] 6 consecutive failures in step encryption-at-rest
- [main] 6 failures in test testPreallocateNonExistingDirectory (2.2% fail rate in 273 executions)
- [main] 6 failures in step encryption-at-rest (100.0% fail rate in 6 executions)
- [main] 6 failures in pipeline elasticsearch-periodic (100.0% fail rate in 6 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/libs/native/src/test/java/org/elasticsearch/nativeaccess/PreallocateTests.java
+++ b/libs/native/src/test/java/org/elasticsearch/nativeaccess/PreallocateTests.java
@@ -9,6 +9,7 @@
 package org.elasticsearch.nativeaccess;
 
 import org.elasticsearch.test.ESTestCase;
+import org.junit.Before;
 
 import java.io.IOException;
 import java.nio.file.Path;
@@ -17,8 +18,14 @@ import java.util.OptionalLong;
 import static org.hamcrest.Matchers.equalTo;
 
 public class PreallocateTests extends ESTestCase {
-    public void testPreallocate() throws IOException {
+
+    @Before
+    public void setup() {
         assumeFalse("no preallocate on windows", System.getProperty("os.name").startsWith("Windows"));
+        assumeFalse("preallocate not supported on encrypted block devices", "encryption-at-rest".equals(System.getenv("BUILDKITE_LABEL")));
+    }
+
+    public void testPreallocate() throws IOException {
         Path file = createTempFile();
         long size = 1024 * 1024; // 1 MB
         var nativeAccess = NativeAccess.instance();
@@ -29,7 +36,6 @@ public class PreallocateTests extends ESTestCase {
     }
 
     public void testPreallocateNonExistingFile() {
-        assumeFalse("no preallocate on windows", System.getProperty("os.name").startsWith("Windows"));
         Path file = createTempDir().resolve("intermediate-dir").resolve("test-preallocate");
         long size = 1024 * 1024; // 1 MB
         var nativeAccess = NativeAccess.instance();
@@ -40,7 +46,6 @@ public class PreallocateTests extends ESTestCase {
     }
 
     public void testPreallocateNonExistingDirectory() {
-        assumeFalse("no preallocate on windows", System.getProperty("os.name").startsWith("Windows"));
         Path file = createTempDir().resolve("intermediate-dir").resolve("test-preallocate");
         long size = 1024 * 1024; // 1 MB
         var nativeAccess = NativeAccess.instance();

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -74,9 +74,6 @@ tests:
 - class: org.elasticsearch.xpack.searchablesnapshots.cache.common.CacheFileTests
   method: testCacheFileCreatedAsSparseFile
   issue: https://github.com/elastic/elasticsearch/issues/110801
-- class: org.elasticsearch.nativeaccess.PreallocateTests
-  method: testPreallocate
-  issue: https://github.com/elastic/elasticsearch/issues/110948
 - class: org.elasticsearch.nativeaccess.VectorSystemPropertyTests
   method: testSystemPropertyDisabled
   issue: https://github.com/elastic/elasticsearch/issues/110949

